### ISSUE TITLE
Rave overrides are only applied from the entry bower.json

### ISSUE THREAD

**Issue by phated** [RaveJS/rave#16]
Adding the following to the `bower.json` file of a dependency does not apply the overrides to `dojo` files.

```
"rave": {
    "overrides": {
        "dojo": {
            "moduleType": "amd"
        }
    }
}
```

**Comment by phated** [comment]
Adding them to the main `bower.json` file seems to apply them correctly.

**Comment by briancavalier** [comment]
This actually seems like the right behavior to me ... or at least, sanity-preserving behavior.  It seems scary if any extension can override any other extension's metadata at will.  What if two extensions want to override the metadata of the same package, but in conflicting ways?  I'm sure there are other crazy cases.  Seems like it could all get out of hand pretty quickly.

IMHO, these overrides are temporary things that help bring compatibility until a library author can fix/update his/her library's metadata.  Issuing a pull request to the author seems like a great thing to do when we find these cases, too, especially for stuff that is non-rave-specific like `moduleType`.

**Comment by unscriptable** [comment]
Yah, sorry for misleading you, @phated.  I'm leaning towards what @briancavalier says: "overrides" should be a way for app devs to fix things, not extension authors.  

That said, one way to fix this is to fork dojo, add a proper bower.json, and point to your github fork of dojo in your rave extension.  Then, issue a PR to dojo.

**Comment by unscriptable** [comment]
Thoughts, @phated?

**Comment by phated** [comment]
I agree that the `overrides` should be at the application-level, and just documented it here as it was opposite of what @unscriptable and I discussed in IRC.  I think that the errors being thrown by invalid metadata/configuration need to be better in leading the application developer to correct the problem.

I think there should be enough metadata lying around to give an error message like such

```
"define was undefined in `dojo/main`, you may need to add
"rave": {
    "overrides": {
        "dojo": {
            "moduleType": "amd"
        }
    }
}
to your bower.json"
```

**Comment by unscriptable** [comment]
Fixed via 0.2.0.
